internal/icinga2/api_responses.go

```diff
@@ -91,6 +100,16 @@ type Downtime struct {
 	Service string `json:"service_name"`
 	Author  string `json:"author"`
 	Comment string `json:"comment"`
+
+	// RemoveTime is used to indicate whether a downtime was ended automatically or cancelled prematurely by a user.
+	// It is set to zero time for the former case, otherwise to the timestamp at which time has been cancelled.
+	RemoveTime UnixFloat `json:"remove_time"`
```

Comment:
> This makes the interface to the outside more obvious

Are you serious, which outside? That field is intended to be used within the package and is never supposed to be used elsewhere.

> (you don't need details from a comment within the struct definition)

I'm not sure what that's supposed to mean.

> (the comment in there states "Neither Icinga 2 nor Icinga DB support negative timestamps for downtime" but the type is elsewhere as well

Are you proposing all of these changes just because the comment mentions downtime? This is just an example of something that I know for sure will cause problems. IMHO, you are focusing on totally non-essential things, after all this code is only temporary as far as I know and we could usefully invest the time elsewhere.

Reply:
> > This makes the interface to the outside more obvious
> 
> Are you serious, which outside? That field is intended to be used within the package and is never supposed to be used elsewhere.
>
> > (you don't need details from a comment within the struct definition)
> 
> I'm not sure what that's supposed to mean.

Outside of this type. The following line could be `else if d.WasCanceled()` instead, making the information from the comment for the member obvious from the type's interface:

https://github.com/Icinga/icinga-notifications/blob/8c380a19ff156daab10dbd3dcc35939db885e8e0/internal/icinga2/client.go#L208

> > (the comment in there states "Neither Icinga 2 nor Icinga DB support negative timestamps for downtime" but the type is elsewhere as well
> 
> Are you proposing all of these changes just because the comment mentions downtime? This is just an example of something that I know for sure will cause problems. IMHO, you are focusing on totally non-essential things, after all this code is only temporary as far as I know and we could usefully invest the time elsewhere.

I found the special-casing in the `UnixFloat` to be a bit strange from the beginning, i.e. that's not really something you'd expect from such a type:

https://github.com/Icinga/icinga-notifications/blob/8c380a19ff156daab10dbd3dcc35939db885e8e0/internal/icinga2/api_responses.go#L28-L32

And now I've got a better idea how to do this instead. Wasn't the whole point of this so that you can write `d.RemoveTime.Time().IsZero()` instead of `d.RemoveTime.Time().Equal(time.UnixMilli(0))` in the `else if` mentioned above?

Also, `UnixFloat` might be a candidate for being moved to our common Go library like `UnixMilli`.